### PR TITLE
User story 33

### LINKED ISSUE #34: User Story 33
Image

As an admin,
When I visit an admin invoice show page (/admin/invoices/:invoice_id)
Then I see information related to that invoice including:
- Invoice id
- Invoice status
- Invoice created_at date in the format "Monday, July 18, 2019"
- Customer first and last name

### PR DESCRIPTION
**Pull request by westonio**
# Describe the changes below:
- Add test, controller actions, and view for Admin Invoices Show page
- Add Invoices model method for concatenating the customer's first and last name
# Relevant user story:
```
33. Admin Invoice Show Page
As an admin,
When I visit an admin invoice show page (/admin/invoices/:invoice_id)
Then I see information related to that invoice including:
- Invoice id
- Invoice status
- Invoice created_at date in the format "Monday, July 18, 2019"
- Customer first and last name
```

# Before submitting, check the following:
- [x] Entire test suite is passing
- [100%] SimpleCov test percentage

### CODE REVIEW

File: app/models/invoice.rb [hunk 1]
```diff
@@ -7,4 +7,8 @@ class Invoice < ApplicationRecord
   validates_presence_of :status
 
   enum :status, { "in progress" => 0, "cancelled" => 1, "completed" => 2 }
+
+  def customer_name
```
Comment (awiley33) [inline app/models/invoice.rb:11]: ooo i like this method to dry up your code!

File: app/views/admin/invoices/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<%= render partial: "shared/admin_topper" %>
+<h3 class="full-length-header"> Invoice #<%= @invoice.id  %></h3>
+<p>Status: <%= @invoice.status.capitalize %></p>
+<p>Created on: <%= @invoice.created_at.strftime('%A, %e %b %Y') %></p>
```
Comment (awiley33) [inline app/views/admin/invoices/show.html.erb:4]: nice use of stringify

Comment (dani-wilson) [inline app/views/admin/invoices/show.html.erb:4]: Nice use of display logic 😻

File: spec/models/invoice_spec.rb [hunk 3]
```diff
@@ -14,6 +14,11 @@
   end
 
   describe 'Instance Methods' do
-    
+    it 'concatenates its customers names' do
+      joey = Customer.create!(first_name: 'Joey', last_name: 'Ondricka')
```
Comment (dani-wilson) [inline spec/models/invoice_spec.rb:18]: Looks super clean!

File: spec/features/admin/invoices/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+      it "shows the invoice's created_at date in the format 'Monday, July 18, 2019'" do
+        expect(page).to have_content(Date.today.strftime('%A, %e %b %Y'), count: 1)
+      end
+
+      it "shows the invoice's Customer first and last name" do
+        expect(page).to have_content(@invoice_1.customer_name, count: 1)
+        expect(page).to_not have_content(@invoice_3.customer.first_name)
+      end
+    end
```
Comment (awiley33) [inline spec/features/admin/invoices/show_spec.rb:57]: feature testing looks robust! good job

**Review by dani-wilson — APPROVED** [review]
Nice work! Looks great

File: spec/models/invoice_spec.rb [hunk 5]
```diff
@@ -14,6 +14,11 @@
...
+      invoice_1 = joey.invoices.create!()
+
+      expect(invoice_1.customer_name).to eq("Joey Ondricka")
+    end
```
Comment (awiley33) [inline spec/models/invoice_spec.rb:22]: nice model instance method test

**Review by awiley33 — APPROVED** [review]
Good work Weston!

**Pull request merged**